src/modules/rewards/components/RewardsActivitiesListItem.tsx

```diff
@@ -44,48 +45,69 @@ const RewardsActivitiesListItem: FC<RewardActivitiesListItemProps> = ({ userId,
   const [errorMessage, setErrorMessage] = useState('');
 
   const RewardsActivityIcon = (type: ActvityType) => {
-
-    if (type === "follow_push_on_discord") {
-      return <Discord width={48} height={48} />
+    if (type === 'follow_push_on_discord') {
+      return <Discord width={48} height={48} />;
     }
 
     if (type === 'follow_push_on_twitter') {
-      return <Twitter width={48} height={48} />
+      return <Twitter width={48} height={48} />;
     }
 
-    // return <PushDefaultRewards />
-    return <RewardsActivity />
-    // return <Discord />
-  }
+    return <RewardsActivity />;
+  };
+
+  const resolveActivityTitle = (activityTitle: string) => {
+    const regex = /\[([^\]]+)\]\(([^)]+)\)/;
+    const match = activityTitle?.match(regex);
+    if (match) {
+      const preText = activityTitle.substring(0, match.index);
+      const linkedText = match[1];
+      const url = match[2];
+      let postText;
+      if (match.index) {
+        postText = activityTitle.substring(match.index + match[0].length);
+      }
+
+      return (
+        <Box display="flex" gap='s1'>
+          <Text variant="bl-semibold"> {preText}</Text>
+          <Link color='pink-500' to={url} target="_blank" rel="noopener noreferrer">
+            <Text variant="bl-semibold" color='pink-500'>{linkedText}</Text>
+          </Link>
+          <Text variant="bl-semibold"> {postText}</Text>
+        </Box>
+      );
+    }
+    return (
+      <Text variant="bl-semibold" color={{ light: 'gray-1000', dark: 'gray-100' }}>
+        {activityTitle}
+      </Text>
+    );
+  };
```

Comment:
**SEPARATE COMPONENT** in a **SEPARATE FILE** for this.

Name: RewardsActivityTitle.tsx

```

const RewardsActivityTitle=({title})=>{

    const extractedTitle = getRewardsActivityTitle(title)
    if(extractedTitle){
        return <Box>...</Box>
    }

    return <Text>...<Text/>

}

```